addon/views/modal.js

```diff
@@ -33,6 +34,23 @@ export default Em.View.extend(
     this.set('visible', true);
   },
 
+  /* Hide with Escape */
+  turnOnEscapeEvent: function() {
+    var _this = this;
+
+    this.set('keydownListener', function(event) {
+      if (event.which == _this.escapeKeyCode) {
+        _this.get('controller').send('closeModal', _this.get('outlet'));
+      }
+    });
+
+    Em.$('body').on('keydown', this.get('keydownListener'));
```

Comment:
I tried registering the event in the view but it never triggered unless i put an input there. It seems like normal elements (except from body) don't trigger keyboard events.